### DISCUSSION TITLE
state change of remote domain should be reflected in button states

### DISCUSSION THREAD

**Discussion by scyto** [home-assistant/frontend#8221] [Dashboard]
I use a harmony remote, it is extremely useful to know that the state of the activities is 'off' this state is provided by the harmony integration. Why?  because i need to know my projector based activity is not on and burning bulbs.

given we have state_color: true/false  for buttons I would argue knowing that your harmony activity (aka a scene - collection of devices) is set in off (aka no activities running) is super useful.

I can see in https://github.com/home-assistant/frontend/pull/4510 that it was decided remote would not change (https://github.com/home-assistant/frontend/pull/4510#issuecomment-576098212).  My ask would be to maintain that default behavior (as it is what most want) unless the state_color: true is set for a button.

**Comment by spacegaier** [comment]
I am not sure I understand what you want to achieve. You want to color the remote icon? A remote itself cannot be on or off?

You mention your projector here. Does the remote know what the projector is on in the Harmony integration? 

Could you provide a few more details for someone who does not own such a remote?

**Reply by scyto** [reply]
Sure.

the best way to think of this is the harmony remote control and hub is more like an automation controller with a remote that has scene buttons. In fact the version I have has no IR on the remote at all - it is all in the hub.

The harmony  hub can control multiple devices including TV, receivers, consoles, smart things devices, hue lights, etc.

These are grouped into activities - aka what to set all the devices to that the harmony knows about (think of it like a scene).

Only one scene can be active at a time.  

The current HA integration surfaces an on and off state  on the entity where:

- on = a scene is active
- off = no scene is active (this state is called PowerOff)

In this example from top to bottom.

I retrieved the current activity in a markup card (this was mostly for testing and learning markdown)
I added the remote entity to an entity card (you can see it has a functioning toggle switch for on/off)
I added the remote entity to a toggle button (and you can see the state as on)
(ignore the activity buttons / device logos, i am not asking for any functionality relative to them)

![image](https://user-images.githubusercontent.com/11369180/105643049-6927ca80-5e42-11eb-83ae-85cb3bd0264a.png)

in this second shot i clicked the remote button i created and you can see the state changed on the button and the entity card.

![image](https://user-images.githubusercontent.com/11369180/105643097-c15ecc80-5e42-11eb-854d-9d07c3e3ae47.png)

what i want to do is eliminate the entity card with the on/off slide toggle and keep just the button, i want to remove the state word on the button to make the UI clean, the problem is then there is no easy way to show the state on that remote button. (i will look at templating but that syntax is on the border of what i am able to do)

I hope that makes sense, given the number of harmony users out there i would expect almost all of them want this functionality....but maybe not.
 
tl;dr so i agree any normal remote has no such thing as on/off - this is the exception to the rule i guess :-)

**Reply by scyto** [reply]
yaml code for the first part of the screenshot if useful

```
type: vertical-stack
title: remote control
cards:
  - type: markdown
    content: >-
      <center> Current Activity: <b>  {{
      states.remote.basement_harmony.attributes["current_activity"] }} <center>
      <b>
  - type: entities
    entities:
      - entity: remote.basement_harmony
  - type: grid
    columns: 3
    cards:
      - type: button
        tap_action:
          action: toggle
        entity: remote.basement_harmony
        show_name: false
        show_state: true
      - type: button
        icon: 'mdi:microsoft-xbox'
        tap_action:
          action: call-service
          service: remote.turn_on
          service_data:
            activity: Play Xbox
            entity_id: remote.basement_harmony
etc
```

**Reply by scyto** [reply]
> You mention your projector here. Does the remote know what the projector is on in the Harmony integration?

It doesn't know. The remote just assumes it is based on whether a scene is active where the projector was sent 'turn on'; if a different scene is selected that doesn't involve the project a 'turn off' is automatically sent .

The reason i have 4 of these hub based remotes is because they are freaking the most reliable IR device i have ever owned, the IR macros never fail and it's bluetooth and wifi control of AV device is reliable too.

(here it is - wow the price on these is insane, i bought all mine for $60 each back in 2014.... https://www.amazon.com/Logitech-Harmony-Companion-Control-Entertainment/dp/B00N3RFC4G)

**Reply by spacegaier** [reply]
@scyto I am not completely sure I get the problem. In the button settings you can disable the rendering of the state label. The button icon should change tot he active color (default yellow) if the underlying entity is in state "on". If that is not the case, then probably due to the fact that the state seems to be "On" (capital O) judging by your screenshots. Is that the value that the integration is setting?

**Reply by scyto** [reply]
@spacegaier  

To answer your question yes the integration is setting the underlying states as follows:

![image](https://user-images.githubusercontent.com/11369180/107160774-2b9c6480-694d-11eb-97da-b5b4108e9e60.png)

producing this 
![image](https://user-images.githubusercontent.com/11369180/107160758-132c4a00-694d-11eb-967d-1f8d8c159b98.png)
(where what i want is that icon to show as yellow)

the problem (i believe) is the remote domain is not a supported domain as per this code https://github.com/home-assistant/frontend/blob/753804f463da2918d7d93556e330fd74458292fc/src/common/style/icon_color_css.ts which i think documents the domains that will toggle the icon between two colors in the frontend.

 Today the button NEVER changes color no matter what I do (but may be i am being stupid :-) )

this is the yaml fragment for that remote entity

```
  - type: entities
    entities:
      - entity: remote.living_room_harmony
        state_color: true
``` 
i should add that what i actually want to do is get rid the toggle and create a button entity

such as
```
- type: button
        tap_action:
          action: toggle
        entity: remote.living_room_harmony
        show_name: false
        show_state: true
```
